qa/integration-tests/src/test/java/io/camunda/it/client/ProcessQueryTest.java

```diff
@@ -0,0 +1,558 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.it.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.qa.util.cluster.TestStandaloneCamunda;
+import io.camunda.zeebe.client.ZeebeClient;
+import io.camunda.zeebe.client.api.command.CreateProcessInstanceCommandStep1;
+import io.camunda.zeebe.client.api.response.*;
+import io.camunda.zeebe.client.api.response.Process;
+import io.camunda.zeebe.client.api.search.filter.ProcessInstanceFilter;
+import io.camunda.zeebe.client.api.search.response.ProcessInstance;
+import io.camunda.zeebe.client.protocol.rest.ProcessInstanceVariableFilterRequest;
+import io.camunda.zeebe.qa.util.junit.ZeebeIntegration;
+import io.camunda.zeebe.qa.util.junit.ZeebeIntegration.TestZeebe;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ZeebeIntegration
+public class ProcessQueryTest {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ProcessQueryTest.class);
+
+  private static final List<Process> DEPLOYED_PROCESSES = new ArrayList<>();
+  private static final List<ProcessInstanceEvent> PROCESS_INSTANCES = new ArrayList<>();
+
+  @TestZeebe
+  private static TestStandaloneCamunda testStandaloneCamunda = new TestStandaloneCamunda();
```

Comment:
`TestStandaloneCamunda` may need to initialized like [here](https://github.com/camunda/camunda/pull/22042/files#diff-543034a3d59ad61aeb751f4ec7b256d4f974d85254bd299c2aae7347a91de8da), but this can be done after https://github.com/camunda/camunda/pull/22042 is merged